File: code/perfect-java-code/perfect-design-pattern/src/creation/factory/simplefactory/pizzastore/pizza/Pizza.java
package creation.factory.simplefactory.pizzastore.pizza;

/**
 * @author Kelly
 * @create 2020-09-18 16:28
 */
public abstract class Pizza {
    protected String name;
    // 准备原材料
    public abstract void prepare();

    public void bake() {
        System.out.println(name + " baking...");
    }

    public void cut() {
        System.out.println(name + " cutting...");
    }

    public void box() {
        System.out.println(name + " boxing...");
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: code/perfect-java-code/perfect-scala/src/main/scala/com/learn/chapter17/simplefactory/pizzastore/pizza/Pizza.scala
package com.learn.chapter17.simplefactory.pizzastore.pizza

/**
 * @author Kelly
 * @create 2021-05-13 15:27
 */
abstract class Pizza {
  var name: String = _

  // 抽象方法
  def prepare()

  def cut(): Unit ={
    println(this.name + " cutting...")
  }

  def bake(): Unit = {
    println(this.name + " baking...")
  }

  def box(): Unit = {
    println(this.name + " boxing...")
  }

}
